JobCancellationException in preview

### Steps

1. Preview OBJ or MTL file with textures.

### Result

Numerous `JobCancellationException`s appear in error log.

### Expected Result

The cause of the exception should be determined and fixed.

Alternatively, the exception should be handled/suppressed, since it doesn't cause any other issues.

### IDE Details

IntelliJ IDEA 2025.1 EAP (Ultimate Edition)
Build #IU-251.17181.16, built on January 22, 2025
Source revision: ab61c6ad10675
Licensed to IntelliJ IDEA EAP user: Slawomir Czerwinski
Expiration date: February 21, 2025
Runtime version: 21.0.5+8-b792.48 amd64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.awt.windows.WToolkit
Windows 11.0
Kotlin plugin: K2 mode
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 8192M
Cores: 16
Registry:
  analyze.exceptions.on.the.fly=true
  ide.experimental.ui=true
Non-Bundled Plugins:
  GLSL (1.24)
  com.jetbrains.space (251.17181.16)
  it.czerwinski.intellij.wavefront (1.4.9-alpha.1)
  DevKit (251.17181.16)
  com.intellij.ideolog (251.17181.16)
  it.czerwinski.ide.plugins.valve (1.1.4)
  com.intellij.properties.bundle.editor (251.17181.16)
  com.github.camork.fileExpander (2.5)
  GrepConsole (13.2.0-IJ2023.3)
  com.intellij.bigdatatools.core (251.17181.16)
  git-extended-update-index (0.1.4)
  com.intellij.bigdatatools.kafka (251.17181.16)
Kotlin: 251.17181.16-IJ

### Stack Trace

Exact property of `MtlMaterialImpl` varies. Here's one example:

```
Unable to render preview: kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@2e957e8a
com.intellij.openapi.progress.CeProcessCanceledException: kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@2e957e8a
	at com.intellij.openapi.progress.Cancellation.ensureActive(Cancellation.java:58)
	at com.intellij.openapi.progress.impl.CoreProgressManager.doCheckCanceled(CoreProgressManager.java:159)
	at com.intellij.openapi.progress.ProgressManager.checkCanceled(ProgressManager.java:336)
	at com.intellij.openapi.progress.ProgressIndicatorProvider.checkCanceled(ProgressIndicatorProvider.java:41)
	at com.intellij.psi.impl.source.tree.CompositeElement.getPsi(CompositeElement.java:699)
	at com.intellij.psi.impl.source.SourceTreeToPsiMap.treeElementToPsi(SourceTreeToPsiMap.java:15)
	at com.intellij.psi.impl.source.tree.SharedImplUtil.getFirstChild(SharedImplUtil.java:37)
	at com.intellij.extapi.psi.ASTDelegatePsiElement.getFirstChild(ASTDelegatePsiElement.java:89)
	at com.intellij.psi.util.PsiTreeUtil.getChildrenOfTypeAsList(PsiTreeUtil.java:413)
	at it.czerwinski.intellij.wavefront.lang.psi.impl.MtlMaterialImpl.getRoughnessMapList(MtlMaterialImpl.java:152)
	at it.czerwinski.intellij.wavefront.lang.psi.impl.MtlMaterialElementImpl.getRoughnessMapElement(MtlMaterialElementImpl.kt:69)
	at it.czerwinski.intellij.wavefront.lang.psi.impl.MtlMaterialElementImpl.getRoughnessChannel(MtlMaterialElementImpl.kt:104)
	at it.czerwinski.intellij.wavefront.editor.gl.ObjPreviewScene.renderFacesPBR(ObjPreviewScene.kt:278)
	at it.czerwinski.intellij.wavefront.editor.gl.ObjPreviewScene.renderFaces(ObjPreviewScene.kt:179)
	at it.czerwinski.intellij.wavefront.editor.gl.ObjPreviewScene.renderModel(ObjPreviewScene.kt:167)
	at it.czerwinski.intellij.wavefront.editor.gl.PreviewScene.doRender(PreviewScene.kt:300)
	at it.czerwinski.intellij.wavefront.editor.gl.BaseScene.onRender(BaseScene.kt:144)
	at graphics.glimpse.ui.GlimpsePanel$GlimpseEventListener.display(GlimpsePanel.kt:81)
	at jogamp.opengl.GLDrawableHelper.displayImpl(GLDrawableHelper.java:692)
	at jogamp.opengl.GLDrawableHelper.display(GLDrawableHelper.java:674)
	at com.jogamp.opengl.awt.GLJPanel$Updater.display(GLJPanel.java:1478)
	at com.jogamp.opengl.awt.GLJPanel$9.run(GLJPanel.java:1553)
	at jogamp.opengl.GLDrawableHelper.invokeGLImpl(GLDrawableHelper.java:1293)
	at jogamp.opengl.GLDrawableHelper.invokeGL(GLDrawableHelper.java:1147)
	at com.jogamp.opengl.awt.GLJPanel$OffscreenBackend.doPaintComponent(GLJPanel.java:2127)
	at com.jogamp.opengl.awt.GLJPanel.paintComponent(GLJPanel.java:587)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:964)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1133)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:964)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1133)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:964)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1133)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:964)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1133)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:964)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1133)
	at java.desktop/javax.swing.JLayeredPane.paint(JLayeredPane.java:586)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:964)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1133)
	at java.desktop/javax.swing.JComponent.paintToOffscreen(JComponent.java:5319)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBufferedImpl(RepaintManager.java:1680)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(RepaintManager.java:1655)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paint(RepaintManager.java:1592)
	at java.desktop/javax.swing.BufferStrategyPaintManager.paint(BufferStrategyPaintManager.java:281)
	at java.desktop/javax.swing.RepaintManager.paint(RepaintManager.java:1352)
	at java.desktop/javax.swing.JComponent._paintImmediately(JComponent.java:5267)
	at java.desktop/javax.swing.JComponent.paintImmediately(JComponent.java:5077)
	at java.desktop/javax.swing.JComponent.paintImmediately(JComponent.java:5089)
	at com.intellij.openapi.ui.LoadingDecorator$LoadingLayer.setAlpha(LoadingDecorator.kt:241)
	at com.intellij.openapi.ui.LoadingDecorator._init_$lambda$0(LoadingDecorator.kt:69)
	at com.intellij.openapi.ui.LoadingLayerAnimator.paintNow(LoadingDecorator.kt:261)
	at com.intellij.util.ui.Animator.paint(Animator.kt:85)
	at com.intellij.util.ui.Animator.onTick(Animator.kt:57)
	at com.intellij.util.ui.Animator.access$onTick(Animator.kt:16)
	at com.intellij.util.ui.Animator$resume$1.invokeSuspend(Animator.kt:131)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:225)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:117)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:43)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:728)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:750)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:582)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:480)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12$lambda$11$lambda$10$lambda$9(IdeEventQueue.kt:307)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:857)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12$lambda$11$lambda$10(IdeEventQueue.kt:306)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$2$lambda$1(IdeEventQueue.kt:948)
	at com.intellij.openapi.application.WriteIntentReadAction.lambda$run$0(WriteIntentReadAction.java:24)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWriteIntentReadAction$lambda$6(AnyThreadWriteThreadingSupport.kt:266)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWithTemporaryThreadLocal(AnyThreadWriteThreadingSupport.kt:217)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWriteIntentReadAction(AnyThreadWriteThreadingSupport.kt:266)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteIntentReadAction(ApplicationImpl.java:927)
	at com.intellij.openapi.application.WriteIntentReadAction.compute(WriteIntentReadAction.java:55)
	at com.intellij.openapi.application.WriteIntentReadAction.run(WriteIntentReadAction.java:23)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$2(IdeEventQueue.kt:945)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$3(IdeEventQueue.kt:959)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:109)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:959)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12(IdeEventQueue.kt:301)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:341)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@2e957e8a
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -20,6 +20,8 @@
 
 ### Fixed
 
+- Ignore `ProcessCanceledException` in error log
+
 ### Security
 
 ## [1.4.8] - 2024-11-10

--- a/src/main/kotlin/it/czerwinski/intellij/wavefront/editor/gl/BaseScene.kt
+++ b/src/main/kotlin/it/czerwinski/intellij/wavefront/editor/gl/BaseScene.kt
@@ -143,10 +143,9 @@ abstract class BaseScene(
             gl.glClearColor(background.toVec3())
             gl.glClear(ClearableBufferType.COLOR_BUFFER, ClearableBufferType.DEPTH_BUFFER)
             doRender(gl)
+        } catch (@Suppress("IncorrectCancellationExceptionHandling") ignored: ProcessCanceledException) {
         } catch (expected: Throwable) {
-            if (expected !is ProcessCanceledException) {
-                onRenderError(gl, expected)
-            }
+            onRenderError(gl, expected)
         } finally {
             if (isStarted) pause()
         }

--- a/src/main/kotlin/it/czerwinski/intellij/wavefront/editor/gl/BaseScene.kt
+++ b/src/main/kotlin/it/czerwinski/intellij/wavefront/editor/gl/BaseScene.kt
@@ -18,6 +18,7 @@ package it.czerwinski.intellij.wavefront.editor.gl
 
 import com.intellij.openapi.Disposable
 import com.intellij.openapi.editor.colors.EditorColorsManager
+import com.intellij.openapi.progress.ProcessCanceledException
 import com.intellij.openapi.progress.util.BackgroundTaskUtil
 import com.intellij.openapi.project.Project
 import com.jogamp.opengl.GLAnimatorControl
@@ -143,7 +144,9 @@ abstract class BaseScene(
             gl.glClear(ClearableBufferType.COLOR_BUFFER, ClearableBufferType.DEPTH_BUFFER)
             doRender(gl)
         } catch (expected: Throwable) {
-            onRenderError(gl, expected)
+            if (expected !is ProcessCanceledException) {
+                onRenderError(gl, expected)
+            }
         } finally {
             if (isStarted) pause()
         }

